### ISSUE TITLE
Google Code-In (2018) Pending issues

### ISSUE THREAD

**Issue by CodeSarthak** [publiclab/plots2#4581]
This is a staging area. Each task has (or will soon have) a link to a full issue including relevant lines of code, screenshots, and other helpful information as we prepare them for solving. Some are harder than others, but those tagged with `first-timers-only` are great to begin with. (more [documentation on tasks](https://developers.google.com/open-source/gci/help/org-sample-tasks))

Add new items: Look here for new items to adapt for this list: [planning issues](https://github.com/publiclab/plots2/labels/planning), [break-me-up issues](https://github.com/publiclab/plots2/labels/break-me-up), [fto-candidates](https://github.com/publiclab/plots2/labels/fto-candidate). Fit tasks into the categories below.

Format items to be descriptive, `[bracket otherwise]`: As we create this list, it's OK to be brief, but if a task lacks description, tags, screenshots, and code links, `[Put it in brackets to indicate that it needs to be fleshed out]`

Break out items here into their own issues: Use the `gci-candidate` tag instead of the `gci` tag to manage the supply. We don't want to have more open issues than we can support at one time, and `gci` is what we'll direct students towards. `gci-candidate` is for "held back" issues we're saving up.

Weighting: some issues don't result in many lines of code, but are still big projects. Use tags `small`, `medium`, `big` when you'd like to help mark them ([read more](https://github.com/publiclab/plots2/issues/3284#issuecomment-422071860))

- [x] Remove asterisk from Boolean Full-Text Searches ([#3490](https://github.com/publiclab/plots2/issues/3490))

-   [x]  Turn off ip_address records in impressionist gem, switch to request_hash [#3382](https://github.com/publiclab/plots2/issues/3382)
-   [ ]  display author profile bio below content in moderator email notifications: <https://github.com/publiclab/plots2/blob/master/app/views/admin_mailer/notify_node_moderators.html.erb>
-   [x]  move files from <https://github.com/publiclab/plots2/tree/master/app/views/comment/> to <https://github.com/publiclab/plots2/tree/master/app/views/comments/> to consolidate these very similar folders, and find any references to `/comment/` to change to `/comments/` -- issue at [#3339](https://github.com/publiclab/plots2/issues/3339)

-   [x]  Making a moving forward link or button on <https://publiclab.org/stats> page (described in [#2650](https://github.com/publiclab/plots2/issues/2650)) -- `GCI dashboard`
-   [x]  Search Auto-complete not working on <https://publiclab.org/post> page (described in [#2795](https://github.com/publiclab/plots2/issues/2795)) -- `GCI dashboard`
-   [x]  Views and number of comments of post is not seen on small screen (described in [#2397](https://github.com/publiclab/plots2/issues/2397)) -- `GCI dashboard`
-   [x]  Change navbar links to icons (described in [#3328](https://github.com/publiclab/plots2/issues/3328) ) --- `GCI dashboard`
-   [x]  Add dropdown icons adjacent to navbar links where required (described in [#3324](https://github.com/publiclab/plots2/issues/3324) ) -- `GCI dashboard`

-   [x]  Trending topics on dashboard should not have repetitions. [#3188](https://github.com/publiclab/plots2/issues/3188) -- GCI dashboard
-   [ ]  complete 'sortable headers' implementation following pattern in [#2352](https://github.com/publiclab/plots2/issues/2352)
-   [ ]  additional work on the Checklists system: [#2452](https://github.com/publiclab/plots2/issues/2452)
-   [ ]  calculate "answered" rate for questions posted in a certain time period (described in [#2120](https://github.com/publiclab/plots2/issues/2120))
-   [x]  Add a footer to the main page [publiclab/image-sequencer#335](https://github.com/publiclab/image-sequencer/issues/335)
-   [x]  Display an installation message to user in the CLI [publiclab/image-sequencer#338](https://github.com/publiclab/image-sequencer/issues/338)
-   [x]  Add a success prompt for saved sequence [publiclab/image-sequencer#342](https://github.com/publiclab/image-sequencer/issues/342)
-   [x]  Add a success prompt for module installation from npm [publiclab/image-sequencer#343](https://github.com/publiclab/image-sequencer/issues/343)
-   [x]  Display a flash message on removal of a step [publiclab/image-sequencer#345](https://github.com/publiclab/image-sequencer/issues/345)
-   [x]  Include a back-to-top Button on scrolling down [publiclab/image-sequencer#346](https://github.com/publiclab/image-sequencer/issues/346)
-   [x]  'No file chosen' wrongly placed in Import Image module [publiclab/image-sequencer#356](https://github.com/publiclab/image-sequencer/issues/356)
-   [x]  Arrange the modules in alphabetical order [publiclab/image-sequencer#375](https://github.com/publiclab/image-sequencer/issues/375)
-   [x]  Rename omniauth.md [#3731](https://github.com/publiclab/plots2/issues/3731) `GCI dashboard`
-   [x]  Allow upload of only image files [publiclab/image-sequencer#429](https://github.com/publiclab/image-sequencer/issues/429)

These can be moved up once they have code links or issue links:

-   [ ]  `needs-code-links` Preserve redirect to original destination when accessing `/post` while not logged in (follow these steps: 1. publiclab.org/post You will be redirected to login page 2. Login 3. Redirected to dashboard -- Expected Behavior Redirected to /post.)
-   [ ]  display of an SVG at the URL `/profile/status/_ID_` for a `user` with `id` of `_ID_`, showing the `user.status` as one of these two SVGs: <https://gist.github.com/jywarren/ea0523a09f6f8660486b14eeea15de95>
-   [x]  display if someone's been banned or is still active in moderator emails (<https://github.com/publiclab/plots2/blob/master/app/views/admin_mailer/notify_node_moderators.html.erb>) using the SVG at the above URL so moderators can tell if something's already been moderated without opening a link -- issue at [#3338](https://github.com/publiclab/plots2/issues/3338)
-   [ ]  make username/email form appear side by side with Oauth user registration at <https://publiclab.org/signup>
-   [x]  `needs-code-links` Remove extra space at bottom of /post\
    [![image](https://user-images.githubusercontent.com/20972099/45542650-2300dc00-b830-11e8-89df-d66847e43dad.png)](https://user-images.githubusercontent.com/20972099/45542650-2300dc00-b830-11e8-89df-d66847e43dad.png)
-   [x]  `needs-code-links` Removal of 1 horizontal line -- issue at #4611 
    [![image](https://user-images.githubusercontent.com/20972099/45542757-665b4a80-b830-11e8-9fca-d432dcff0397.png)](https://user-images.githubusercontent.com/20972099/45542757-665b4a80-b830-11e8-9fca-d432dcff0397.png)
-   [ ]  `needs-code-links` Add generic video icon here, instead of being specific to YouTube. We should give equal priority to all the hosts. (<https://github.com/publiclab/PublicLab.Editor/>)\
    [![image](https://user-images.githubusercontent.com/20972099/45542852-a9b5b900-b830-11e8-9101-b5085c6565d3.png)](https://user-images.githubusercontent.com/20972099/45542852-a9b5b900-b830-11e8-9101-b5085c6565d3.png)
-   [ ]  On clicking on '?' page should be opened in new tab. Currently, it is opening on the same page, using `target="_blank"` on the `<a>` tag\
    [![image](https://user-images.githubusercontent.com/20972099/45542912-d79afd80-b830-11e8-95d9-4bbfb533c1cd.png)](https://user-images.githubusercontent.com/20972099/45542912-d79afd80-b830-11e8-95d9-4bbfb533c1cd.png) -- `GCI dashboard`
-   [ ]  `needs-code-links` Similar behaviour (needs to open on a new page) reported for `Open Source Your Work` link at <https://publiclab.org/signup> and /post
-   [ ]  `needs-code-links` Hover text should redirect to page listing all the wiki articles and research notes with text `water quality`\
    [![image](https://user-images.githubusercontent.com/20972099/45543039-35c7e080-b831-11e8-8d25-bc060801a490.png)](https://user-images.githubusercontent.com/20972099/45543039-35c7e080-b831-11e8-8d25-bc060801a490.png)
-   [ ]  `needs-code-links`Hover text should redirect to page listing all the wiki articles and research notes with text `balloon-mapping`\
    [![image](https://user-images.githubusercontent.com/20972099/45543048-382a3a80-b831-11e8-8f1b-e10dc81ee4de.png)](https://user-images.githubusercontent.com/20972099/45543048-382a3a80-b831-11e8-8f1b-e10dc81ee4de.png)
-   [x]  `needs-code-links` Redundant `Dashboard` in the drop down menu [#3327](https://github.com/publiclab/plots2/issues/3327)\
    [![image](https://user-images.githubusercontent.com/20972099/45543183-96571d80-b831-11e8-85e9-722515ea1e56.png)](https://user-images.githubusercontent.com/20972099/45543183-96571d80-b831-11e8-85e9-722515ea1e56.png)
-   [ ]  `needs-code-links` Clearly mention that on RHS we have Research notes. `Liked by ____` appears ambiguous to me a lot\
    [![image](https://user-images.githubusercontent.com/20972099/45543333-ec2bc580-b831-11e8-82f1-4e3e216d1914.png)](https://user-images.githubusercontent.com/20972099/45543333-ec2bc580-b831-11e8-82f1-4e3e216d1914.png)
-   [ ]  `needs-code-links` sort wiki page likes according to date/time\
    [![image](https://user-images.githubusercontent.com/20972099/45543424-06fe3a00-b832-11e8-9ec9-32e564306ee6.png)](https://user-images.githubusercontent.com/20972099/45543424-06fe3a00-b832-11e8-9ec9-32e564306ee6.png)
-   [ ]  `needs-code-links` If a person has no barnstars and they click on the nav bar. There must be a message to show them. Just to enhance the readability\
    [![image](https://user-images.githubusercontent.com/20972099/45543539-5d6b7880-b832-11e8-8b2b-33cdf3c28ebf.png)](https://user-images.githubusercontent.com/20972099/45543539-5d6b7880-b832-11e8-8b2b-33cdf3c28ebf.png)
-   [ ]  On removal of any of these tags, all three must be removed [![image](https://user-images.githubusercontent.com/20972099/45543633-9c99c980-b832-11e8-9e50-0c52fa207118.png)](https://user-images.githubusercontent.com/20972099/45543633-9c99c980-b832-11e8-9e50-0c52fa207118.png)
-   [ ]  Same button three times to post work on same web page [![image](https://user-images.githubusercontent.com/20972099/45563088-e4881300-b869-11e8-87e2-c9108fb6f5c9.png)](https://user-images.githubusercontent.com/20972099/45563088-e4881300-b869-11e8-87e2-c9108fb6f5c9.png)
-   [ ]  Same thread is placed repeatedly on the dashboard page due to multiple comments.\
    [![image](https://user-images.githubusercontent.com/20972099/45541563-1464f580-b82d-11e8-98ca-c5b7759389ad.png)](https://user-images.githubusercontent.com/20972099/45541563-1464f580-b82d-11e8-98ca-c5b7759389ad.png)\
    I will suggest that one post should be shown only once.
-   [x]  Use of common functions in /maps page : [#3316](https://github.com/publiclab/plots2/issues/3316)
-   [ ]  Standardize display of `n` Wiki and `m` Research Notes [#2193](https://github.com/publiclab/plots2/issues/2193)
-   [x]  When a person submits an empty sign up form.\
    [![image](https://user-images.githubusercontent.com/20972099/44982930-14473900-af95-11e8-8588-f84b35cb0491.png)](https://user-images.githubusercontent.com/20972099/44982930-14473900-af95-11e8-8588-f84b35cb0491.png)\
    Error should also come that the fields are empty [#3340](https://github.com/publiclab/plots2/issues/3340) -- `GCI dashboard`
-   [ ]  Test for the the above task
-   [ ]  Tag the author on the home page where their articles are written in grid view/List view (needs own issue)
-   [ ]  logging users to wiki page when a user logs in via header on wiki page (needs own issue)
-   [ ]  Clicking on like button if not logged in should redirect to login page. (needs own issue)
-   [ ]  Pagination broken on home page\
    [![image](https://user-images.githubusercontent.com/20972099/45540558-62c4c500-b82a-11e8-92a9-53dfe375afbc.png)](https://user-images.githubusercontent.com/20972099/45540558-62c4c500-b82a-11e8-92a9-53dfe375afbc.png)\
    see that url shows page 2 but the pagination is of page 1. (needs own issue)
-   [ ]  `needs-code-links` make tags in `Suggested tags` on publiclab.org/post appear in tag input box when clicked - <https://github.com/publiclab/PublicLab.Editor/>

Documentation/Training: creating/editing documents and helping others learn more
--------------------------------------------------------------------------------

-   [x]  Write a page describing different tags used at github repositories like GCI, outreach, etc [#2341](https://github.com/publiclab/plots2/issues/2341) -- `GCI dashboard`
-   [x]  write a document explaining how to start contributing at public lab for Google Summer Of Code [#3319](https://github.com/publiclab/plots2/issues/3319)-- `GCI dashboard`
-   [ ]  write a document explaining how to start contributing at Public Lab for RGSoc (needs own issue)
-   [x]  write documentation on how to open a Pull Request for first - timer contributor (needs own issue)
-   [ ]  the table of contents on the github.com/publiclab/plots2 is inconsistent with the content present after it (needs own issue)
-   [x]  Add a list of available modules to the documentation [publiclab/image-sequencer#337](https://github.com/publiclab/image-sequencer/issues/337)

Outreach/Research: community management, marketing, or studying problems and recommending solutions
---------------------------------------------------------------------------------------------------

-   [x]  Reporting Bugs on PL /Reporting Suggestions [#3460](https://github.com/publiclab/plots2/issues/3460) -- `GCI dashboard`
-   [x]  Create new `first-timers-only` issues (<https://publiclab.org/n/13667>) -- `GCI dashboard`
-   [x]  Convert `fto-candidates` into `first-timers-only` issues by embedding the friendly message which you can find in `first-timers-only` issues. `fto-candidate` issues can be found at <https://github.com/publiclab/plots2/labels/fto-candidate> -- `GCI dashboard`
-   [ ]  embed our welcome page in full at <https://publiclab.org/code> (needs own issue)
-   [ ]  read through welcome page <https://publiclab.github.io/community-toolbox/> and highlight gaps or ambiguities (needs own issue in <https://github.com/publiclab/community-toolbox/issues>)
-   [x]  Refactor <https://publiclab.github.io/community-toolbox/> to ensure list of contributors shows on multi-repository page (complex) - [publiclab/community-toolbox#29](https://github.com/publiclab/community-toolbox/issues/29)
-   [x]  Help add code links to existing issues (described in [#3424](https://github.com/publiclab/plots2/issues/3424)) -- `GCI dashborad`
-   [x]  Update welcome message in a pull request with asking a twitter handle of that user (described in [#3485](https://github.com/publiclab/plots2/issues/3485)) -- `GCI dashboard`
-   [x]  Find a bug -- Google code-in [publiclab/image-sequencer#350](https://github.com/publiclab/image-sequencer/issues/350)

Quality Assurance: testing and ensuring code is of high quality
---------------------------------------------------------------

-   [ ]  [#2151](https://github.com/publiclab/plots2/issues/2151) - Fix "Show 0 more" on inline people lists
-   [x]  [#2150](https://github.com/publiclab/plots2/issues/2150) - `Show more` should change to `Show less` on clicking -- GCI dashboard
-   [ ]  Read through Code Climate "smells" and open issues related to solving them with code links included and recommendations: <https://codeclimate.com/github/publiclab/plots2/issues?severity%5B%5D=critical&severity%5B%5D=major&status%5B%5D=&status%5B%5D=open&status%5B%5D=confirmed&engine_name%5B%5D=structure&engine_name%5B%5D=duplication> (described in [#3706](https://github.com/publiclab/plots2/issues/3706)) -- `GCI dashboard`.

Design: graphic design or user interface design
-----------------------------------------------

-   [x]  Exploring layouts for the dashboard [#3355](https://github.com/publiclab/plots2/issues/3355) instance count 50
-   [ ]  BIG TASK: Wiki page should be rendered on same page. On clicking on wiki pages nav bar on /notes/popular a new page opens up. But wiki pages should be rendered under the wiki nav bar. [#3456](https://github.com/publiclab/plots2/issues/3456)
-   [ ]  FTO: make the rsvp links in the events on the home page non clickable.
-   [x]  add hover text in [public lab editor](https://github.com/publiclab/PublicLab.Editor/)\
    [![image](https://user-images.githubusercontent.com/20972099/44983562-3d68c900-af97-11e8-98a7-e6bf22956075.png)](https://user-images.githubusercontent.com/20972099/44983562-3d68c900-af97-11e8-98a7-e6bf22956075.png) [publiclab/PublicLab.Editor#156](https://github.com/publiclab/PublicLab.Editor/issues/156)
-   [ ]  [![image](https://user-images.githubusercontent.com/20972099/45532373-6bf56800-b811-11e8-9be0-ba0aa16d1905.png)](https://user-images.githubusercontent.com/20972099/45532373-6bf56800-b811-11e8-9be0-ba0aa16d1905.png)\
    ''Read More'' should be center aligned
-   [ ]  "Read more" and the next grid item should have hr in between\
    [![image](https://user-images.githubusercontent.com/20972099/45532417-92b39e80-b811-11e8-9f86-b69bc9b417c6.png)](https://user-images.githubusercontent.com/20972099/45532417-92b39e80-b811-11e8-9f86-b69bc9b417c6.png)
-   [x]  Implement "grid view" for inline lists, in addition to existing "table/list view" - [#1097](https://github.com/publiclab/plots2/issues/1097)
-   [x]  some Bootstrap 4 pre-upgrade work - [#1716](https://github.com/publiclab/plots2/issues/1716)
-   [ ]  [multiple-user]Copy welcomebot and first-timers-only templates into other projects - [#3681](https://github.com/publiclab/plots2/issues/3681) -- `GCI dashboard`

**Comment by CodeSarthak** [comment]
Hi guys @publiclab/gci-18 @ValentinaTironi @kevinzluo @okonek , let us first figure which of these issues already have their separate issues. 

If they do, then comment in this format : 

(Name of Checklist Instance) (Issue Number)

and tick the checkbox.

After figuring out which tasks are already made, we can work on making new issues for the rest. 

This can be also be a good source for first timers and those interested in documentation. (@SidharthBansal @jywarren )

**Comment by oorjitchowdhary** [comment]
I've ticked on issues which are already solved..

**Comment by CodeSarthak** [comment]
> I've ticked on issues which are already solved..

Goof job, mate 👍

**Comment by CodeSarthak** [comment]
Hi, @publiclab/guides @publiclab/reviewers ,

I am not sure but I believe that this issue is still relevant and we can add issues from this list to GCI too. Though, the ones which are already solved need to be closed (39 of 77 mentioned are not yet checked).

Thank you.

**Comment by SidharthBansal** [comment]
If you want to help us you can go ahead. However these will not be counted
against points.
Thanks Sarthak

**Comment by CodeSarthak** [comment]
Yes, of course. I was not looking to do this for points, just for the record. It seems that it would be quite helpful to the community. 

I will try to work on this as and when I get time.

**Comment by SidharthBansal** [comment]
I can upload a three point task for refactoring this to 2019 task list when
you will complete it.
Thanks for enthu

**Comment by CodeSarthak** [comment]
Sure, thank you for the support @SidharthBansal .

**Comment by CodeSarthak** [comment]
Hey @jywarren @SidharthBansal ,

I have tried my level best here to look for already completed tasks and I have come to the conclusion that 33 issues are still there which can be resolved, some of which are also ftos and some are multiple-use too (for example, #3681 ). 

This can prove to be a good source of ftos and even GCI tasks. I will get to work immediately on creating issues for the ones which don't already have one (about 15 so).

Thank you.

**Comment by SidharthBansal** [comment]
Superb

**Comment by CodeSarthak** [comment]
Hey @SidharthBansal , can you upload a GCI task for this refactoring?

**Comment by SidharthBansal** [comment]
You can do other tasks right now.
I will upload it on Tuesday. Ping me if I forget.
I am busy preparing for my exam tomorrow 😣

**Comment by CodeSarthak** [comment]
Yes, sure. All the best for your exam :laughing:

**Comment by SidharthBansal** [comment]
Please describe what you did and how can i check your work. Please help :-)

**Comment by CodeSarthak** [comment]
I went through the issues/tasks list above and figured out which of them are still unresolved. I checked the ones which have been resolved.

You can check the same by going through the issues checklist above. 

I am planning to add issues for some of the tasks above which do not have an issue of their own. Though this may require help from the community since it is a large number. 

Thank you.

**Comment by SidharthBansal** [comment]
Hello Sarthak,
Please move the unsolved issues to GCI 2019-20 from GCI 2018-19. Please
close this issue once done.
Thanks

**Comment by CodeSarthak** [comment]
Hey, 

I have opened a new issue (https://github.com/publiclab/plots2/issues/7014)  with only the issues/tasks which were left. I removed all those which has already been closed or completed. 

Thus, this issue is now redundant. 

Thank you.
